### ISSUE TITLE
[BUG] Hardware decoding not working, works fine in Chrome

### ISSUE THREAD

**Issue by miromarszal** [hmlendea/gfn-electron#208]
### Describe the bug

I see no signs of hardware decoding when streaming a game. There is a noticeable input lag too.

### To Reproduce

1. Start gfn-electron and launch your favourite game.
2. Wait until the stream begins.
3. Observe your GPU usage with an appropriate tool (`intel_gpu_top` in my case).

### Screenshots

GPU usage with gfn-electron
![GPU usage with the app](https://github.com/hmlendea/gfn-electron/assets/29063081/b892d588-cea6-435c-a70d-760ce5a49196)

GPU usage with Chrome
![GPU usage with Chrome](https://github.com/hmlendea/gfn-electron/assets/29063081/315768e5-d8d0-4a1b-80f4-9083ce3275cf)

### Operating System

Ubuntu 23.04

### Desktop Environment

GNOME 44.3

### Display Server

Wayland

### Installation method

Source

### Version

1.13.0

### Is this a fresh install of the app or an update from a past version?

Fresh

### Did this issue appear right away upon installation/updating, or spontaneously?

Right after installation

### Additional context

The graphics card is Intel® UHD Graphics (CML GT2). Hardware decoding works for me in Chrome with only one additional flag added: `-enable-features=VaapiVideoDecodeLinuxGL`. I leave out `--use-gl`, however, I verified that `egl` does not work and `angle` does. These are not the same as used by gfn-electron, so I tried the default setting and flags set as above by tweaking the source, in both cases without success. You can see from the screenshot above that hardware decoding does not work.

**Comment by seedneus** [comment]
didn't it need a permissions set in flatseal? Default is disabled as I can see...

**Comment by miromarszal** [comment]
Good point about flatpak installations, there could be issues there. In my case it's a source installation though.

**Comment by hmlendea** [comment]
> didn't it need a permissions set in flatseal? Default is disabled as I can see...

Which permission are you refering to?

**Comment by hmlendea** [comment]
Is this still an issue in the latest release?

**Comment by erik-mueller** [comment]
Yes, I can confirm that this issue still exists. Starting chromium with --enable-accelerated-video-decode works, however this option is not accepted by the geforcenow-electron binary. Is it somehow possible to provide custom arguments to geforcenow-electron?

**Comment by chrispouliot** [comment]
@hmlendea Was the argument passing added in the open PR able to fix this issue? I am wondering if I can use AV1 decode with this but I am hesitant to pay for the higher subscription that offers it if hardware decode is not working. Thanks!

**Comment by ErebosGR** [comment]
Any progress with this? Is it a limitation of electron not enabling GPU acceleration?
Without hardware video decoding, GFN is practically unusable.

**Comment by ErebosGR** [comment]
> On the Limitation? Not any more I believe, at least when i updated electron and change a few things it started working. [try this AppImage build](https://github.com/AscendXP/gfn-electron-with-hardware-acceleration/releases/download/Latest/geforcenow-electron_2.1.0_linux.AppImage).

Thank you, I tried it and so far it works just as well as the Chromium flatpak.
It doesn't even have the issues #250 and #255 .

**Comment by Play-InTheWay** [comment]
> Thank you, I tried it and so far it works just as well as the Chromium flatpak. It doesn't even have the issues [#250](https://github.com/hmlendea/gfn-electron/issues/250) and [#255](https://github.com/hmlendea/gfn-electron/issues/255) that I encountered with the original electron version.
> 
> Nobara/KDE, Wayland.

Appimage “solves” the issue [#250](https://github.com/hmlendea/gfn-electron/issues/250) just because it runs with Xwayland by default, running the flatpak one on Xwayland also fixes it. You can try to run the appimage with the “--enable-features=UseOzonePlatform --ozone-platform-hint=auto" flags, and that error would still happen.

However, when I try to run GeForce Now through Brave in native Wayland, that error doesn't happen. So the error could be fixable, but until then, the Flatpak version shouldn't enable Wayland by default.

**Comment by ErebosGR** [comment]
> Appimage “solves” the issue [#250](https://github.com/hmlendea/gfn-electron/issues/250) just because it runs with Xwayland by default, running the flatpak one on Xwayland also fixes it. You can try to run the appimage with the “--enable-features=UseOzonePlatform --ozone-platform-hint=auto" flags, and that error would still happen.

I didn't know that, thank you for the insight.

> However, when I try to run GeForce Now through Brave in native Wayland, that error doesn't happen. So the error could be fixable, but until then, the Flatpak version shouldn't enable Wayland by default.

I think the Chromium flatpak also runs on Wayland, and I already said that the issue doesn't appear there.
I think the issue is that Electron is missing certain parts from the Chromium engine.
